### PR TITLE
implemented Permissions, then solved a bug on the recorded audio file…

### PR DESCRIPTION
**Pull request by mkenlo**
…Path: changed the file path for startRecorder method
Then added permissiosn for storage and microphone. Now before using the mic the app makes sure user has allow access to microphone and  storage.

### CODE REVIEW

File: lib/screens/translation_list_screen.dart [hunk 1]
```diff
@@ -42,7 +42,7 @@ class _TranslationListScreenState extends State<TranslationListScreen> {
           Translation item = data[index];
           return ListTile(
             title: Text(item.sentence),
-            subtitle: Text("Oct 20, 2019"),
+            subtitle: Text(item.recordedOn),
```
Comment (evindj) [inline lib/screens/translation_list_screen.dart:45]: +1

File: lib/screens/translation_screen.dart [hunk 2]
```diff
@@ -68,30 +73,27 @@ class TranslationScreenState extends State<TranslationScreen> {
...
+        this._isRecording = true;
+      });
+      print(path);
+      _recorderSubscription = _flutterSound.onRecorderStateChanged.listen((e) {
+        if (e == null) {
+          return;
+        }
+        DateTime date =
+            new DateTime.fromMillisecondsSinceEpoch(e.currentPosition.toInt());
+        String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
```
Comment (evindj) [inline lib/screens/translation_screen.dart:88]: Check if formating does not already provide the final string. So you don't have to do substring afterwards.

File: lib/services/permission_service.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return permissionStatus == PermissionStatus.granted;
+  }
+
+
+  Future<bool> _requestPermission(List<PermissionGroup> permissions) async {
+    var result = await _permissionHandler.requestPermissions(permissions);
+    bool granted = true;
+    result.forEach((permission, permissionStatus) {
+      if (permissionStatus != PermissionStatus.granted)
+        granted = granted && false;
```
Comment (evindj) [inline lib/services/permission_service.dart]: how about we get rid of the if statement  ```granted = granted && permissionStatus == PermissionStatus.granted```

**Review by evindj — CHANGES_REQUESTED** [review]

File: lib/services/permission_service.dart (hunk 3 above)
Comment (mkenlo) [reply lib/services/permission_service.dart]: This is actually elegant! Thanks

File: lib/screens/translation_screen.dart (hunk 2 above)
Comment (mkenlo) [reply lib/screens/translation_screen.dart:88]: No the formatting does not provide the final string. This is an example of what you get when formatting "00:05:437"

**Pull request merged**